src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +16,40 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t throw country to DB!", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Country country;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            country = session.get(Country.class, id);
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Can`t get country from the DB by id: " + id, e);
```

Comment:
The exception message should be specific to the operation. Consider changing 'Can`t get country from the DB by id: ' to 'Can't retrieve country with ID: ' to make it clearer.